plugins/scanners/fossid/src/main/kotlin/FossId.kt

```diff
@@ -962,6 +982,61 @@ class FossId internal constructor(
                         requests += async {
                             service.markAsIdentified(config.user, config.apiKey, scanCode, path, false)
                         }
+
+                        val filteredSnippetChoicesByPath = snippetChoices.filter {
+                            it.given.sourceLocation.path == path
+                        }
+
+                        val relevantSnippetChoices = filteredSnippetChoicesByPath.filter {
+                            it.choice.reason == SnippetChoiceReason.ORIGINAL_FINDING
+                        }
+                        relevantSnippetChoices.forEach { filteredSnippetChoice ->
+                            val match = SNIPPET_PURL_REGEX.matchEntire(filteredSnippetChoice.choice.purl.orEmpty())
+                            match?.let {
```

Comment:
```kotlin
SNIPPET_PURL_REGEX.matchEntire(filteredSnippetChoice.choice.purl.orEmpty())?.also { match ->
  ...
}
```

(`also` instead of `let` as the return value of the closure is not used)